lib/masksets.gd

```diff
@@ -0,0 +1,55 @@
+#############################################################################
+##
+#W  masksets.gd                  GAP library                     Steve Linton
+##
+##
+#Y  Copyright (C) 2017 The GAP Group
+##
+##  This file declares the operations for mask sets.
+##
+
+
+
+#############################################################################
+##
+#F  MakeMaskSet(<width>[, <width>[, <width>...]]])
+##  <#GAPDoc Label="MakeMaskSet">
+##  <ManSection>
+##  <Func Name="MakeMaskSet" Arg='width....'/>
+##
+##  <Description>
+##
+##  This function sets up the machinery for a set of bitfields of the given
+##  widths. The total of the widths must not exceed 60 bits on 64-bit architecture
+##  or 28 bits on a 32-bit architecture and the number of widths must not exceed
+##  10 on a 64-bit aerchitecture or 6 on a 32-bit architecture. Also for performance 
+##  reasons some checks that one might wish to do are ommitted. In particular, 
+## the builder and setter functions do not check if the value[s] passed to them are 
+## negative or too large.
+##
+##  You can tell which architecture you are running on by acccessing 
+##  <C>GAPInfo.BytesPerVariable</C> which is 8 on 64-bits and 4 on 32. 
+##
+##  The return value  when <M>n</M> widths are given is a record whose fields are 
+##  <List>
+##  <Mark><C>widths</C></Mark> <Item>a copy of the arguments, for convenience,</Item>
+##  <Mark><C>builder</C></Mark> <Item> a function of <M>n</M> arguments that returns an immediate integer 
+##                 into which they are all packed. The first field occupies the
+##                 least significant bits, etc.</Item>
+##  <Mark><C>getters</C></Mark> <Item> a list of <M>n</M> functions of one argument each of which extracts 
+##                 one of the fields from an immediate integer</Item>
+##  <Mark><C>setters</C></Mark> <Item> a list of <M>n</M> functions of two arguments each of which returns the packed value
+##                 in which one of the fields has been replaced by the given value
+##                 note that this does NOT modify the immediate integer.</Item>
```

Comment:
It is not immediately clear what "the" immediate integer here is. Perhaps say: "... a list of n functions, each taking two arguments,  <packedValue> and <x>; the return value of the i-th setter is a new packed value, with the value of the i-th field set to <x>."

All in all, I think it would be a good idea to pick a specific set of terms for the various "parts" involved here, and stick to them. E.g. referring to the same thing sometimes as "immediate integer" and sometimes as "packed value" is confusing. I'd explain the connection once, then stick to one term.